src/screens/AddVocab/AddVocab.jsx

```diff
@@ -1,198 +1,285 @@
+  }, []);
+
+  useEffect(() => {
+    if (!selectedPackage) {
+      return;
+    }
+
+    const grps = packages.find((p) => p.id === selectedPackage.value);
+
+    setGroups(() => {
+      if (!grps) {
+        return [];
+      }
+
+      return grps.Groups;
+    });
+
+    setSelectedGroup((grp) => {
+      if (!grp) {
+        return null;
+      }
+
+      const contain = grps.Groups.find((g) => g.id === grp.value);
+
+      if (!contain) {
+        return null;
+      }
+
+      return grp;
     });
-  } else {
-    groupDropdownItems = [];
-  }
+  }, [packages, selectedPackage]);
+
+  useEffect(() => {
+    setPackageItems(() =>
+      packages.map((p) => {
+        const foreignIcon = languages.find(
+          (x) => x.name === p.foreignWordLanguage
+        ).icon;
+        const translatedIcon = languages.find(
+          (x) => x.name === p.translatedWordLanguage
+        ).icon;
+
+        return {
+          value: p.id,
+          label: (
+            <CustomSelectOption
+              name={p.name}
+              postfix={foreignIcon + " - " + translatedIcon}
+            />
+          ),
+        };
+      })
+    );
+  }, [packages]);
+
+  useEffect(() => {
+    setGroupsItems(() =>
+      groups.map((g) => {
+        return {
+          value: g.id,
+          label: g.name,
+        };
+      })
+    );
+  }, [groups]);
 
   return (
-    <div className="add-vocab-form">
-      <h1 className="heading">{t("screens.addVocab.title")}</h1>
+    <>
+      <div className="add-vocab-form">
+        <h1 className="heading">{t("screens.addVocab.title")}</h1>
 
-      <div className="dropdowns">
-        <div className="select-wrapper">
-          <Select
+        <div className="dropdowns">
+          <div className="select-wrapper">
+            <Select
+              required
+              creatable
+              createText={"Create new Package"}
+              onCreate={openPackageModal}
+              tabIndex={1}
+              label={t("global.package")}
+              options={packageItems}
+              onChange={(v) => {
+                setSelectedPackage(v);
+              }}
+              value={selectedPackage}
+              noOptionsMessage={t("screens.addVocab.noPackagesMessage")}
+            />
+          </div>
+          <div className="select-wrapper">
+            <Select
+              required
+              creatable
+              createText={"Create new Group"}
+              onCreate={openGroupModal}
+              disabled={!selectedPackage}
+              tabIndex={1}
+              label={t("global.group")}
+              options={groupsItems}
+              onChange={(v) => {
+                setSelectedGroup(v);
+              }}
+              value={selectedGroup}
+              noOptionsMessage={t("screens.addVocab.noGroupsMessage")}
+            />
+          </div>
+        </div>
+        <div className="input-fields">
+          <TextInput
```

Comment:
we also need a active checkbox and an description field. See: vocascan/vocascan-server#32

Reply:
I would like to use the switcher for the active state.